docs/architecture/adr-015-crypto-encoding.md

```diff
@@ -35,9 +35,9 @@ increasing the space attached to each validator / account.
 
 The alternative to using amino here would be for us to create an enum type.
 Switching to just an enum type is worthy of investigation post-launch.
-For referrence, part of amino encoding interfaces is basically a 4 byte enum
+For reference, part of amino encoding interfaces is basically a 4 byte enum
 type definition.
-Enum types would just change that 4 bytes to be a varuint, and it would remove
+Enum types would just change that 4 bytes to be a variant, and it would remove
```

Comment:
I believe varuint is correct

Reply:
i went looking for a definition to link to, came to here: https://developers.google.com/protocol-buffers/docs/encoding#signed-integers but not sure if its correct. @ValarDragon do you have something to link here?

Reply:
It should be a `varint`, my bad!

This is something thats slightly confusing, but `varints` do support the full unsigned range. The way they handle negatives is to just take the two's complement encoding, and treat it as a uint of the same size.

Reply:
thank you 😄